keyboards/handwired/ms_sculpt_mobile/keymaps/milestogo/keymap.c

```diff
@@ -121,146 +164,308 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-
-float tone_startup[][2]    = SONG(STARTUP_SOUND);
-float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
-float tone_colemak[][2]    = SONG(COLEMAK_SOUND);
-#endif
-
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
-  switch (keycode) {
-    case QWR:
-      if (record->event.pressed) {
-        #ifdef AUDIO_ENABLE
-          PLAY_SONG(tone_qwerty);
-        #endif
-        layer_off(_CDH);
-      }
-      return false;
-      break;
-
-    case CDH:
-      if (record->event.pressed) {
-        #ifdef AUDIO_ENABLE
-          PLAY_SONG(tone_colemak);
-        #endif
-        layer_on(_CDH);
-      }
-      return false;
-      break;
-
-    case SYM:
-      if (record->event.pressed) {
-        layer_on(_SYM);
-      } else {
-		layer_off(_SYM);
-      }
-     return false;
-     break;
-
-   }
- return true;
-
-}
-
-const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
-{
-
-/* If id is in the range of BABL macros, call the babl function */
-/* Any clever remapping with modifiers should happen here e.g. shift bablkey does opposite*/
-#ifdef USE_BABLPASTE
-
-   if( id >= BABL_START_NUM && id < (BABL_START_NUM + BABL_NUM_MACROS ) ) {
-   		if (record->event.pressed)  { // is there a case where this isn't desired?
-
-   			babblePaste ( record,  id );
-   			return MACRO_NONE;
-   		}
-   	}
-#endif
-
+    switch (keycode) {
+        case QWR:
+            if (record->event.pressed) {
+                layer_off(_CDH);
+            }
+            return false;
+            break;
 
-  // MACRODOWN only works in this function
-      switch(id) {
-        case 0:
-          if (record->event.pressed) {
-            register_code(KC_RSFT);
-          } else {
-            unregister_code(KC_RSFT);
-          }
-        break;
+        case CDH:
+            if (record->event.pressed) {
+                layer_on(_CDH);
+            }
+            return false;
+            break;
 
- /* Colemak mod-dh moves the D key to the qwerty V position
-   This hack makes apple-V_position do what I mean */
- 	case DHPASTE:
-		if(keyboard_report->mods & MOD_BIT(KC_LGUI) ) {
-			if (record->event.pressed) {
-	  	 		clear_keyboard_but_mods();
-		 		register_code(KC_V);
-	     	} else {
-		 		unregister_code(KC_V);
-	    	}
-		} else {
+        case SYM:
             if (record->event.pressed) {
-          		register_code(KC_D);
+                layer_on(_SYM);
             } else {
-           		unregister_code(KC_D);
-	   		}
-		}
-	break;
-
-	case VIBRK: // vi esc:
-		 if (record->event.pressed) {
-			return MACRO( T(ESC),D(LSFT),T(SCLN),U(LSFT), END );
-		 }
-	break;
-
-
-
-
-	default:
-    	return MACRO_NONE;
+              layer_off(_SYM);
+            }
+            return false;
+            break;
+
+        /* Colemak mod-dh moves the D key to the qwerty V position
+            This hack makes apple-V_position do what I mean */
+        case DHPASTE:
+            if(keyboard_report->mods & MOD_BIT(KC_LGUI) ) {
```

Comment:
I believe the"proper" way to get the mods is now: 
```suggestion
            if(get_mods() & MOD_BIT(KC_LGUI) ) {
```

IIRC, I think the keyboard_report method may sometimes give outdated information.